rspace/src/test/scala/coop/rchain/rspace/history/TimingMeasure.scala

```diff
@@ -0,0 +1,180 @@
+        if (nonEmptyItemsCount > 0 && itemsReady < nonEmptyItemsCount) {
+          val itemIndex = startIdx % itemsCount
+          val updatedNode = if (itemIndex % 2 > 0) {
+            node.updated(
+              itemIndex,
+              Leaf(ByteVector(bytes(startIdx % 128)), ByteVector(bytes(size = 32)))
+            )
+          } else {
+            node.updated(
+              itemIndex,
+              NodePtr(ByteVector(bytes(startIdx % 128)), ByteVector(bytes(size = 32)))
+            )
+          }
+
+          generateNode(updatedNode, itemsReady + 1, startIdx + 1)
+        } else node
+
+      val averageNum    = 50
+      val averageWarmUp = 1
+
+      val result =
+        (0 until (averageNum + averageWarmUp)).toList
+          .foldLeft((0L, 0L)) {
+            case ((timeEncode, timeDecode), i) =>
+              val nodes = (0 until num).map { numNode =>
+                generateNode(emptyNode, itemsReady = 0, numNode)
+              }
+
+              def statistic(timeEncode: Long, timeDecode: Long): Unit = {
+                def iI(v: Int) = "%7d) ".format(v)
+
+                def mS(v: Long) = "%7.3f ".format(v.toDouble / 1000)
+
+                val str = iI(i) + "  " + mS(timeEncode) + "    " + mS(timeDecode)
+                println(str)
+              }
+
+              val (nodesEncode, timeEncodeTemp) = {
+                val t0  = System.nanoTime
+                val res = nodes.map(anyCodec.encode)
+                val t1  = System.nanoTime
+                val m   = Duration.fromNanos(t1 - t0).toMillis
+                (res, m)
+              }
+
+              val (nodesDecode, timeDecodeTemp) = {
+                val t0  = System.nanoTime
+                val res = nodesEncode.map(v => anyCodec.decode(v))
+                val t1  = System.nanoTime
+                val m   = Duration.fromNanos(t1 - t0).toMillis
+                (res, m)
+              }
+
+              assert(nodesDecode == nodes, "Encoded and decoded node should be same")
+
+              if (averageStatistic) statistic(timeEncodeTemp, timeDecodeTemp)
+
+              if (i < averageWarmUp) (timeEncode, timeDecode)
+              else
+                (
+                  timeEncode + timeEncodeTemp,
+                  timeDecode + timeDecodeTemp
+                )
+          }
+
+      def numStr(v: Int) = "%7d".format(v)
+
+      def mS(v: Long) = "%7.3f".format(v.toDouble / (1000.toDouble * averageNum.toDouble))
+
+      val str = numStr(num) + " | " + s"${nonEmptyItemsCount} | " +
+        mS(result._1) + " | " + mS(result._2)
+      println(str)
+    }
+
+    def fS(v: String): String = "%7s".format(v)
+
+    println(s"${codecName}")
+    val strTitle = fS("num") + " | " + fS("Non-empty items count") + " | " +
+      fS("timeEnc(msec)") + " | " + fS("timeDec(msec)")
+    println(strTitle)
+
+    (0 until itemsCount).foreach { x =>
+      experiment(num = 5, x)
+    }
+  }
+}
```

Comment:
We are in the process of removing old implementation so this code is not relevant.